pkg/agent/flowexporter/connections/conntrack.go

```diff
@@ -59,7 +57,7 @@ func filterAntreaConns(conns []*flowexporter.Connection, nodeConfig *config.Node
 		// Therefore, we ignore the connection with ClusterIP and keep the connection with Endpoint Pod IP.
 		// Conntrack flows will be different for Pod-to-Service flows w/ Antrea-proxy. This implementation will be simpler, when the
 		// Antrea proxy is supported.
-		if serviceCIDR.Contains(srcIP) || serviceCIDR.Contains(dstIP) {
+		if !nodeConfig.PodCIDR.Contains(dstIP) {
```

Comment:
The comment above states: "One is with ClusterIP as source or destination" but you only check for the destination here. Is the comment inaccurate?

**Also I am not sure the check is correct**: IIRC we only export flow records from the source Node (see connections.go). At the source Node, the `dstIP` will not be in the `nodeConfig.PodCIDR` CIDR unless the destination Pod is itself local. This CIDR is the local PodCIDR for the Node, not the cluster-wide CIDR.

Reply:
Yes, you are right. My intention was to use cluster PodCIDR and not the local one. Overlooked its definition. 
I can stick with serviceCIDR as before or get the subnets of all nodes from k8s client to perform the check. Any suggestions on preference?
The comment needs to be revised.

This warrants more e2e test coverage for flow exporter checking for inter-node case. Will create an issue.

Reply:
I think potentially the logic needs to be different based on whether AntreaProxy is enabled or not:

- if it is not enabled, keep using `serviceCIDR` as it is required in this case
- if it is enabled, the Antrea Agent knows the individual ClusterIPs for all Services IIRC

Reply:
Oh ok. I see why we are on different pages. This check is required only when Antrea Proxy is disabled. We do not see the duplicate connections with Antrea Proxy. 
I will take `isAntreaProxy` flag as input from agent.go and use it here, so that we do this check with serviceCIDR only when Antrea Proxy is disabled.

Reply:
That sounds good. Please make sure the comment is updated to reflect this info.

Reply:
When AntreaProxy is enabled, do we already maintain a ClusterIP map to check?

Reply:
I think Antonin's suggestion was to get the clusterIP list or query the clusterIP from the installed services map maintained in pkg/agent/proxy. As this check is not necessary with AntreaProxy enabled, so we just skip it rather than checking for clusterIP.

Reply:
Ok